src/herder/TransactionQueue.cpp

```diff
@@ -771,69 +603,33 @@ TransactionQueue::ban(Transactions const& banTxs)
         auto stateIter = mAccountStates.find(kv.first);
         if (stateIter != mAccountStates.end())
         {
-            // If there are no transactions in the queue for this source
-            // account, then there is nothing to do
-            auto& transactions = stateIter->second.mTransactions;
-            if (!transactions.empty())
+            auto const& transaction = stateIter->second.mTransaction;
+            // Only ban transactions that are actually present in the queue.
+            // Transactions with higher sequence numbers than banned
```

Comment:
How is this possible? Do we ever ban txs after accepting new ones for the same account?

Reply:
Similarly here, at apply time, we need to reconcile the block contents with the queue, so banned transactions might not overlap with the queue.

Reply:
(note that this assumption has been there prior to this change, but in this PR the logic is rewritten a bit to simplify it)

Reply:
Thanks for the explanation. I think this logic makes sense and should indeed be preserved.